.circleci/config.yml

```diff
@@ -735,13 +728,10 @@ jobs:
     steps:
       - attach_workspace:
           at: /home/circleci/go/src/github.com/influxdata/influxdb
-      - run:
-          name: Log into Quay
-          command: docker login -u "$QUAY_USER" -p $QUAY_PASS quay.io
+      - quay_login
       - run:
           name: Push the image to Quay
           command: |
             docker load < docker-image.tar
             docker tag quay.io/influxdb/oss-acceptance:${CIRCLE_SHA1} quay.io/influxdb/oss-acceptance:latest
-            docker push quay.io/influxdb/oss-acceptance:${CIRCLE_SHA1}
```

Comment:
Is monitor-ci always pulling `latest`? If so, what happens when there are parallel CI runs from our end?

Reply:
Yep it's dropped because it has already happened. That's basically the "candidate" image.

Reply:
@danxmoran hopefully this addresses your question:

When we run the CI from the `influxdb` end, a "candidate" image is tagged with the git sha and pushed to quay in the `build` step. The `e2e-monitor-ci` runs the `run-monitor-ci-tests` script with that sha as an argument, and the script calls the CircleCI API with the sha as part of the payload as well. Then over in the [`monitor-ci` pipeline](https://github.com/influxdata/monitor-ci/blob/18ce39cfc6c0bbdedd190272c789332c3d5c9832/.circleci/config.yml#L236), that sha-tagged image is tested with the latest UI image. So if we have parallel CI runs on the `influxdb` end, they will each test with their own individual images against the latest UI image.

It works conceptually the same going the other way as well. Changes in the `ui` repo run the `monitor-ci` pipeline with the build from that specific `ui` commit against the latest image from `influxdb`, which will be tagged as `oss-acceptance:latest`.

There is a potential condition where for example you might run the tests at some point in time on the `influxdb` end and everything looks good, but between the time you run the tests and merge it in something changes on the `ui` end that breaks whatever you did. The same thing could happen going the other way of course. This would at least show up on the e2e job for the build in `master`. I think the risk is pretty small since it theoretically shouldn't happen unless we make API-breaking changes in `influxdb`, but there's still the potential.

Another good data point - up until very recently, the `ui` side was testing against the `2.0-beta` OSS image. I was able to replace this with the latest image and not have any of the tests break...so it seems like changes to the backend breaking the e2e tests should be pretty rare!

Reply:
It seems that the potential scenario you describe wherein one side tests the other `latest` image, which then sees a breaking change could be mitigated somewhat by logging the commit SHA of whatever `latest` image "from the other side" is being used for the test. Then if the test succeeds, only to fail later, a comparison can be made against the images used, rather than having to try to correlate which image was tagged as `latest` a the time of test.